### PR TITLE
Improve documentation and interface specification

### PR DESCRIPTION
**Pull request by PDLdeLange**
- Renamed scene object pose message to pose_WA to indicate reference frames. Will update all references after review and before I land this issue on master. 
- Shifted environment identifiers by one and reserved first ID for future use, e.g. environmental presets.
- Provided short description of Imaginator.
- Added more comments to specify units, and so on.

### CODE REVIEW

File: ddln_imaginator.proto [hunk 1]
```diff
@@ -4,15 +4,26 @@ package ddln;
 
 import "ddln_common.proto";
 
+// Imaginator interface definitions
+//
+// This file contains interface definitions for Daedalean's cloud image generator: Imaginator.
+// Imaginator is a cloud rendering service which takes image requests and returns generated images.
```
Comment (gyakushev) [inline ddln_imaginator.proto:10]: mention `stateless` somewhere

File: ddln_imaginator.proto [hunk 2]
```diff
@@ -4,15 +4,26 @@ package ddln;
...
 message CameraIntrinsics {
     required float verticalFieldOfView_deg = 1;
 }
 
 message SceneObject {
     optional string type = 1;
-
-    // pose in ENU coordinates
-    optional Pose pose = 2;
+    optional Pose pose_WA = 2; // pose of object A in world frame W (see ddln_common.proto)
```
Comment (gyakushev) [inline ddln_imaginator.proto]: move the comment above the property definition, don't use same-line comments in this file please

File: ddln_imaginator.proto [hunk 3]
```diff
@@ -36,111 +52,90 @@ message Environment {
...
+        optional float altitude_deg = 3; // altitude, ranges 0..90 [deg]
+        optional float intensity = 4; // intensity value, typical range 0..1 [-]
     }
-    optional Sun sun = 4;
+    optional Sun sun = 5;
 
+    // Configure rain particles.
+    // Note 1: The particles are rendered in real time, e.g. subsequent image requests should be made in real time
+    // for realistic rendering.
+    // Note 2: Rain sprites always appear to fall down vertically since camera velocity is not taken into account.
```
Comment (gyakushev) [inline ddln_imaginator.proto:124]: details of the current implementation should be kept out of the interface definition file
here we specify what is the intent of the message

**Review by gyakushev — CHANGES_REQUESTED** [review]

File: ddln_imaginator.proto (hunk 3 above)
Comment (PDLdeLange) [reply ddln_imaginator.proto:124]: This message is indicating an limitation. The user should be informed IMO.

File: ddln_imaginator.proto [hunk 4]
```diff
@@ -25,6 +38,12 @@ enum ImageFormat {
 }
 
 message Environment {
+    reserved 1; // Reserved for future preset configurations, e.g. CLEAR_SUNNY, HEAVY_RAIN, CUSTOM, and so on.
```
Comment (gyakushev) [inline ddln_imaginator.proto]: nit: also move to the line above

File: ddln_imaginator.proto [hunk 5]
```diff
@@ -25,6 +38,12 @@ enum ImageFormat {
...
+
+    // Add multiple layers of clouds to the scene.
+    // Note: cloud movement is rendered in real time and contain an internal state.
```
Comment (gyakushev) [inline ddln_imaginator.proto]: nit: `contain` => `contains`

File: ddln_imaginator.proto [hunk 6]
```diff
@@ -36,100 +55,119 @@ message Environment {
...
+
+        // density value, typical range 0..1 [-]
         optional float density = 3;
 
-        // directional x,y,z values to set the wind in the Engine.
+        // directional x, y, z values to set the wind in ENU coordinates system [m/s]
         optional Point3d velocity = 4;
     }
-    repeated Clouds clouds = 1;
+    repeated Clouds clouds = 2; // Up to 3 layers supported
```
Comment (gyakushev) [inline ddln_imaginator.proto]: same: move to the line above

Comment (gyakushev) [reply ddln_imaginator.proto]: nit: it is nice if all comments are full English sentences: starting with a capital letter, ending with a dot, contain a verb etc.

**Review by gyakushev — APPROVED** [review]
Thanks!

**Pull request merged**